cli/src/main.rs

```diff
@@ -5,26 +5,41 @@
 #![deny(unused_crate_dependencies)]
 
 mod commands;
-use clap::Parser as ClapParser;
+use clap::{
+    crate_authors,
+    crate_version,
+    ColorChoice,
+    Parser as ClapParser,
+    Parser,
+};
 
 /// Subxt utilities for interacting with Substrate based nodes.
 #[derive(Debug, ClapParser)]
-enum Command {
-    Metadata(commands::metadata::Opts),
-    Codegen(commands::codegen::Opts),
-    Compatibility(commands::compatibility::Opts),
-    Version(commands::version::Opts),
+pub enum SubCommand {
+    Metadata(commands::metadata::MetadataOpts),
+    Codegen(commands::codegen::CodegenOpts),
+    Compatibility(commands::compatibility::CompatOpts),
+    Version(commands::version::Version),
+}
+
+#[derive(Debug, Parser)]
+#[clap(version = crate_version!(), author = crate_authors!(), color=ColorChoice::Always)]
+pub struct CliOpts {
+    #[clap(subcommand)]
+    pub subcmd: SubCommand,
```

Comment:
Is there any advantage to these being two structs and not one struct like before now?